utilities/inference_utils.py

```diff
@@ -71,7 +74,10 @@ def get_deployment_type(self) -> str:
         ):
             return deployment_type
 
-        return self.inference_service.instance.status.deploymentMode
+        if isinstance(self.inference_service, InferenceService):
+            return self.inference_service.instance.status.deploymentMode
+
+        return KServeDeploymentType.SERVERLESS
```

Comment:
can you please add here a check if `InferenceGraph` so it is clear that this is returned on InferenceGraph?
btw - why not get it from the InferenceGraph first isvc?

Reply:
> btw - why not get it from the InferenceGraph first isvc?

The InferenceGraph has its own deployment mode. So, for example, an ISVC being in Serverless mode doesn't mean that the IG is also Serverless.

Reply:
> can you please add here a check if `InferenceGraph` so it is clear that this is returned on InferenceGraph?

Done!

Reply:
> The InferenceGraph has its own deployment mode
So we should get it from the InferenceGraph, right?

Reply:
> > The InferenceGraph has its own deployment mode
> So we should get it from the InferenceGraph, right?

Yes. I think this is covered by lines 70-73, no? These `if` statements are a fallback in case the annotation is missing.

In contrast to ISVCs, we still don't have the deployment mode in the status of the InferenceGraph. So we can only rely in the annotation and fallback to some default if the annotation is not present.

Or am I missing something?

Reply:
so line 81 (default return) should be removed. we do not want to assume any deployment mode if we do not hit any of the conditions.
another option (and i can do it in a following pr) is to return the default mode from the card if we do not find any annotation/status. wdyt?